plugin/hooks.go

```diff
@@ -149,4 +154,41 @@ type Hooks interface {
 	// Note that this method will be called for files uploaded by plugins, including the plugin that uploaded the post.
 	// FileInfo.Size will be automatically set properly if you modify the file.
 	FileWillBeUploaded(c *Context, info *model.FileInfo, file io.Reader, output io.Writer) (*model.FileInfo, string)
+
+	// PushNotificationWillBeSent is invoked before the push notification has been sent to the push proxy server.
+	// You should return the notification that you want to be sent to the user (normally the same notification).
+	//
+	// You can reject the notification returning a nil notification.
+	//
+	// You can modify the notification, returning a changed version of the notification.
```

Comment:
For consistency:

```suggestion
	// You can modify the notification, returning a changed version of the notification.
	//
	//  If you don't need to modify or rejected notifications, use PushNotificationWillBeSent instead. 
```

Reply:
You meant `PushNotificationHasBeenSent`? It is not exactly the same because `WillBeSent` is executed every time a push notification is sent to the push proxy, but `HasBeenSent` is only executed if the notification has been sent without errors. It depends on your intentions with the plugin.

Reply:
I see, thanks for the clarification.